tests/test_project_status.py

```diff
@@ -365,6 +367,39 @@ def test_set_project_to_available_valid_transition(module_client, test_project):
     assert db_deadline == calc_deadline
 
 
+def test_set_project_to_available_no_mail(client):
+    """Set status to Available for test project, but skip sending mails"""
+
+    response = client.post(
+        tests.DDSEndpoint.PROJECT_CREATE,
+        headers=tests.UserAuth(tests.USER_CREDENTIALS["unituser"]).token(client),
+        json=proj_data_with_existing_users,
+    )
+    assert response.status_code == http.HTTPStatus.OK
+    assert response.json and response.json.get("user_addition_statuses")
+    for x in response.json.get("user_addition_statuses"):
+        assert "associated with Project" in x
+
+    public_project_id = response.json.get("project_id")
+
+    with unittest.mock.patch.object(flask_mail.Mail, "send") as mock_mail_send:
+        with unittest.mock.patch.object(
+            dds_web.api.user.AddUser, "compose_and_send_email_to_user"
+        ) as mock_mail_func:
+            response = client.post(
+                tests.DDSEndpoint.PROJECT_STATUS,
+                headers=tests.UserAuth(tests.USER_CREDENTIALS["unituser"]).token(client),
+                query_string={"project": public_project_id},
+                json={"new_status": "Available", "deadline": 10, "send_email": False},
+            )
+        # assert that no mail is being sent.
+        assert mock_mail_send.call_count == 0
+        assert mock_mail_func.called == False
```

Comment:
```suggestion
            # assert that no mail is being sent.
            assert mock_mail_func.called == False

        assert mock_mail_send.call_count == 0
```

Reply:
I think it makes sense for the `mock_mail_func` to be asserted inside the `with` statement and not outside of it.

Reply:
Absolutely! Wrong indentation here.... 😳 

I basically sabotaged my own test! 😖